src/OrleansRuntime/Messaging/GatewayAcceptor.cs

```diff
@@ -41,7 +41,10 @@ protected override bool RecordOpenedSocket(Socket sock)
             {
                 //convert handshake cliendId to a GeoClient ID 
                 if (!string.IsNullOrEmpty(Silo.CurrentSilo.ClusterId))
+                {
                     client = GrainId.NewClientId(client.PrimaryKey, Silo.CurrentSilo.ClusterId);
+                    adjustReturnAddressOnNextMessage = true;
```

Comment:
I wonder also what will happen in the unlikely case if the first couple of messages get reordered, so that this code will reset the `SendingGrain` header of the *second* message while the client will look for cluster ID in the reply to the *first* message. Do you think this is possible?